c++/src/kj/mutex.c++

```diff
@@ -210,7 +210,10 @@ Mutex::AcquiredMetadata Mutex::lockedInfo() const {
 Mutex::Mutex(): futex(0) {}
 Mutex::~Mutex() {
   // This will crash anyway, might as well crash with a nice error message.
-  KJ_ASSERT(futex == 0, "Mutex destroyed while locked.") { break; }
+  // Atomic load to make TSAN happy.
```

Comment:
Wait, hold on... if TSAN is alerting on this, doesn't that suggest an actual problem? It suggests that a thread is trying to destroy a Mutex without having actually synchronized with the last thread that unlocked the Mutex. That suggests it's possible that, under the right circumstances, the Mutex could end up being destroyed while still locked.